object/provider.go

```diff
@@ -397,13 +397,17 @@ func providerChangeTrigger(oldName string, newName string) error {
 }
 
 func FromProviderToIdpInfo(ctx *context.Context, provider *Provider) *idp.ProviderInfo {
+	hostUrl := provider.Host
```

Comment:
Just change the code for your provider type like:

```go
if provider.Type == "WeChat" {
		if ctx != nil && strings.Contains(ctx.Request.UserAgent(), "MicroMessenger") {
			providerInfo.ClientId = provider.ClientId2
			providerInfo.ClientSecret = provider.ClientSecret2
		}
	}
```

Reply:
That makes more sense. I just updated the PR. Please check!